### ISSUE TITLE
.NET Core 3.0 with IHostEnvironment

### ISSUE THREAD

**Issue by jasonycw** [NLog/NLog#3166]
In .NET Core 2.X we can use `IHostingEnvironment` to `.ConfigureNLog(...)` which NLog.config file to be used. 
But with .NET Core 3.0, `IHostingEnvironment` will become obsoleted and will become be replaced by `IHostEnvironment` eventually according to https://github.com/aspnet/AspNetCore/issues/7749
https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-2.2&tabs=visual-studio#hostbuilder-replaces-webhostbuilder

Any planing on when will NLog support .NET Core 3.0 ?

**Comment by 304NotModified** [comment]
Thanks for the heads up. 

Could you please help us with a PR?

**Comment by snakefoot** [comment]
@jasonycw Just curious. Which version of NetStandard supports this new `IHostEnvironment` ?

**Comment by 304NotModified** [comment]
Good question, I think 2.1: https://github.com/dotnet/standard/blob/master/docs/versions.md

**Comment by snakefoot** [comment]
So NetStandard2.1 with dependency on 3.0.0 preview-package: https://www.nuget.org/packages/Microsoft.Extensions.Hosting.Abstractions/ ?

When a single platform dependends on a preview-package, then the entire package also becomes a preview-package. Not sure we are going to have NLog.Extension.Logging in preview-stage until NetCore3 is fully ready. Or am I missing something?

**Comment by 304NotModified** [comment]
Yes, I think it makes sense to merge of .net core issue final, or maybe indeed a preview package (but not from master then).

**Comment by snakefoot** [comment]
Yes master (or dev) would not be a good place for preview-features. Since we like to make changes and releases while waiting for NetCore3 to complete.
